### PR TITLE
Update API generator to use new OpenAPI specification

### LINKED ISSUE #580: Update code generator to use new version of OpenAPI specification
The https://github.com/opensearch-project/opensearch-api-specification repository has been updated to be written natively in OpenAPI, rather than being built from Smithy. This means that some structure and details of the specification has changed. The code generator should be updated to use this new version of the specification. Any changes to the generated output should be validated to confirm if the changes are correct by cross referencing with the OpenSearch implementation & documentation. Any errors found in the specification should be reported on the specification repository.

### PR DESCRIPTION
**Pull request by Xtansia**
### Description
Update API generator to use new OpenAPI specification

Fixes #580 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/OpenSearch.Net/Api/Enums.cs [hunk 1]
```diff
@@ -148,17 +148,6 @@ public enum Size
...
-	[StringEnum]
-	public enum Level
-	{
-		[EnumMember(Value = "cluster")]
-		Cluster,
-		[EnumMember(Value = "indices")]
-		Indices,
-		[EnumMember(Value = "shards")]
-		Shards
-	}
```
Comment (dblock) [inline src/OpenSearch.Net/Api/Enums.cs:160]: Is this a breaking API change?

File: src/OpenSearch.Net/_Generated/Api/RequestParameters/RequestParameters.Cluster.cs [hunk 2]
```diff
@@ -281,13 +300,6 @@ public string AwarenessAttribute
             set => Q("awareness_attribute", value);
         }
 
-        /// <summary>Specify the level of detail for returned information.</summary>
```
Comment (dblock) [inline src/OpenSearch.Net/_Generated/Api/RequestParameters/RequestParameters.Cluster.cs:284]: Another potentially breaking change.

**Review by dblock — CHANGES_REQUESTED** [review]
There are a few changes here that looks breaking API-wise, e.g. `HealthStatus` replaces `WaitForStatus`, etc. 

Either bump major version in this PR and add UPGRADING.md, or add backwards compatible code?

**Comment by Xtansia** [comment]
> There are a few changes here that looks breaking API-wise, e.g. `HealthStatus` replaces `WaitForStatus`, etc.
> 
> Either bump major version in this PR and add UPGRADING.md, or add backwards compatible code?

`main` is already developing the next major version (2.x) and includes other breaking changes. I'll make an UPGRADING.md

**Review by dblock — APPROVED** [review]

**Pull request merged**